cli/running/writer.go

```diff
@@ -38,23 +38,26 @@ func init() {
 
 type ColorizedWriter struct {
 	prefix string
+	out    io.Writer
 }
 
 func (w *ColorizedWriter) Write(p []byte) (int, error) {
 	buf := bytes.NewBuffer(p)
 
 	n := 0
 	for {
-		lineBytes, err := buf.ReadBytes('\n')
-		if err == io.EOF {
+		// from the doc
+		// https://golang.org/pkg/bytes/#Buffer.ReadBytes
+		// > ReadBytes returns err != nil if and only if the returned data does not end in delim
+		// so, we read bytes until 0 bytes returned
+		lineBytes, _ := buf.ReadBytes('\n')
+		if len(lineBytes) == 0 {
 			break
 		}
-		if err != nil {
-			return n, err
-		}
 
 		// prefix
-		if _, err := os.Stdout.Write([]byte(w.prefix)); err != nil {
+		if nPrefix, err := w.out.Write([]byte(w.prefix)); err != nil {
+			n += nPrefix
```

Comment:
What is ```n```?

Reply:
`Write` function of `io.Writer` interface returns `n, err`, wrere `n` is number of bytes that was written.